src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1313,6 +1319,25 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly genesetMolecularProfile = remoteData<MolecularProfile | undefined>({
+        await: () => [
+            this.molecularProfilesInStudies
+        ],
+        invoke: () => {
+            const applicableProfiles = _.filter(
```

Comment:
We should move this to a helper function and test it

Reply:
What would be a sensible place for such a helper function? The top level of the same module?

Reply:
I think there's two other things wrong with this function. First, I think the array I intended to use is `selectedMolecularProfiles` instead of `molecularProfilesInStudies`, as I meant to select only queried profiles. And second, it can be a bit confusing and difficult to explicitly typecheck a MobxPromise that resolves with `undefined` since its `.result` property is also undefined if the promise hasn't resolved (and usually presumed not to be after it has). I'll add two commits to address this.